File: backend/src/main/java/reports/reports/domain/FieldOfStudy.java
package reports.reports.domain;

import org.springframework.data.annotation.CreatedBy;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedBy;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import javax.persistence.*;
import java.util.List;

@Entity
@EntityListeners(AuditingEntityListener.class)
public class FieldOfStudy {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String name;
    private String description;
    @CreatedDate
    private Long createdDate;
    @LastModifiedDate
    private Long lastModifiedDate;
    @CreatedBy
    private String createdBy;
    @LastModifiedBy
    private String lastModifiedBy;

    @OneToMany(mappedBy = "fieldOfStudy")
    private List<Subject> subjects;

    public FieldOfStudy() {}

    public FieldOfStudy(String name, String description) {
        this.name = name;
        this.description = description;
    }
}

File: frontend/src/main/frontend/src/app/component/component.admin/fieldofstudy/fieldofstudy.ts
import {Subject} from "../subject/subject";

export class FieldOfStudy {

  id: number;
  name: string;
  description: string;
  createdDate: string;
  lastModifiedDate: string;
  createdBy: string;
  lastModifiedBy: string;
  subjects : Subject[];

  constructor(json? : any) {
    if (json != null) {
      this.id = json.id;
      this.name = json.name;
      this.description = json.description;
      this.createdDate = json.createdDate;
      this.lastModifiedDate = json.lastModifiedDate;
      this.createdBy = json.createdBy;
      this.lastModifiedBy = json.lastModifiedBy;
      if (json.subjects != null) {
        this.subjects = Subject.toArray(json.subjects);
      }
    }
  }

  // Utils
  static toArray(jsons : any[]) : FieldOfStudy[] {
    let fieldsofstudies : FieldOfStudy[] = [];
    if (jsons != null) {
      for (let json of jsons) {
        fieldsofstudies.push(new FieldOfStudy(json));
      }
    }
    return fieldsofstudies;
  }
}
